### PR TITLE
Nicer node-fetch example

### PR DESCRIPTION
**Pull request by kirill-konshin**

### CODE REVIEW

File: example/node-fetch.ts [hunk 1]
```diff
@@ -7,27 +7,22 @@ export async function callBackend({
 }: CallBackendArguments<{}>) {
   const searchPath = query ? `?${new URLSearchParams(query)}` : '';
   const url = `${baseUrl}${path}${searchPath}`;
+  switch (bodyType) {
+    case 'json':
+      headers = {...headers, 'content-type': 'application/json'};
```
Comment (0xR) [inline example/node-fetch.ts]: I prefer not to mutate the header and body types. If you want to improve the fetch example you could create some more `const` declarations to make it more readable.

File: example/node-fetch.ts [hunk 2]
```diff
@@ -7,27 +7,22 @@ export async function callBackend({
...
-        }
-      : { headers }),
+      method,
+      body,
+      headers,
   });
-
   const text = await response.text();
-  if (response.status >= 200 && response.status < 300) {
+  if (response.ok) {
```
Comment (0xR) [inline example/node-fetch.ts]: :+1:

File: example/node-fetch.ts (hunk 1 above)
Comment (kirill-konshin) [reply example/node-fetch.ts]: It's not a mutation. Here I change reference from one object to another, assign operator will do it in the local scope, original user provided arguments won't be mutated. If you want I can edit this.

Comment (kirill-konshin) [reply example/node-fetch.ts]: This rule covers this btw, consider adding it to your ESLint: https://eslint.org/docs/rules/no-param-reassign

Comment (kirill-konshin) [reply example/node-fetch.ts]: Fixed.

**Comment by 0xR** [comment]
I had a go at this as well and came up with #155 
Hopefully it is simplified enough. Let me know otherwise.

Thanks for you contribution!

**Comment by kirill-konshin** [comment]
Good job throwing away my work. Thanks a lot.

**Pull request closed without merging**